Feature/bjs2 78001 test work schedule

Ramil23-bit opened the pull request.

Ramil23-bit committed:
added test for WorkScheduleService and Validation LocalTime

20cognosce commented on src/test/java/school/faang/user_service/service/work_schedule/ServiceWorkScheduleTest.java:
```diff
@@ -0,0 +1,126 @@
+        long userId = 1L;
+        User user = new User();
+        user.setId(userId);
+        user.setWorkSchedule(workSchedule);
```
в поля это вынеси, ID, user-а и его график

20cognosce commented on src/test/java/school/faang/user_service/service/work_schedule/ServiceWorkScheduleTest.java:
```diff
@@ -0,0 +1,126 @@
+            new WorkSchedule(2L, LocalTime.of(8, 0), LocalTime.of(16, 0),
+                    LocalTime.of(11, 0), LocalTime.of(12, 0),
+                    "Europe/London", new User()));
+
```
вот это криво выглядит

20cognosce commented on src/test/java/school/faang/user_service/service/work_schedule/ServiceWorkScheduleTest.java:
```diff
@@ -0,0 +1,126 @@
+        when(workScheduleRepository.save(any(WorkSchedule.class)))
+                .thenAnswer(invocation -> invocation.getArgument(0));
+
+        WorkSchedule createWorkSchedule = workScheduleService.addWorkSchedule(workScheduleList.get(0));
```
Нет, не надо из статического списка так брать
тут у тебя график не переопределяется в каждом тесте, ты сделал что все тесты работают с одним из списка - это плохо, потому что какой-то из них чет в нем изменить, а другие не в курсе

либо пересоздавай новый в BeforeEach либо внутри теста для каждого создавай свой

20cognosce commented on src/test/java/school/faang/user_service/service/work_schedule/ServiceWorkScheduleTest.java:
```diff
@@ -0,0 +1,126 @@
+        when(workScheduleRepository.save(any(WorkSchedule.class)))
+                .thenAnswer(invocation -> invocation.getArgument(0));
+        WorkSchedule saveWorkSchedule = workScheduleService.addWorkSchedule(workScheduleList.get(0));
+        assertNotNull(saveWorkSchedule);
```
в чем разница с тестов выше?

20cognosce commented on src/test/java/school/faang/user_service/service/work_schedule/ServiceWorkScheduleTest.java:
```diff
@@ -0,0 +1,126 @@
+
+        assertEquals(LocalTime.of(9, 0), result.getStartTime());
+        assertEquals(LocalTime.of(14, 0), result.getEndLunch());
+        assertEquals("Asia/Tokyo", result.getTimezone());
```
почему только 3 поля проверил, а остальные?

20cognosce commented on src/test/java/school/faang/user_service/service/work_schedule/ServiceWorkScheduleTest.java:
```diff
@@ -0,0 +1,126 @@
+                workScheduleService.getById(id));
+        String messageException = String.format(String.format("Work schedule not Found %d",
+                        id));
+        String actualMessage = exception.getMessage();
```
где порядок, сначала подготовка, затем выполнение и потом проверки. Пустыми строками отделяй этапы

20cognosce replied:
сообщение не надо проверять

20cognosce commented on src/test/java/school/faang/user_service/validator/work_schedule_validation/WorkScheduleValidationTest.java:
```diff
@@ -0,0 +1,92 @@
+        LocalTime startLunch = workSchedule.getStartLunch();
+
+        assertTrue(startWork.isAfter(startLunch) || startWork.equals(startLunch),
+                "Start to work after start to Lunch ");
```
не надо это писать, тем более одинаковое в каждом тесте

Ramil23-bit committed:
fixed code

removed the list from the fields, removed the extra test, deleted the message from WorkScheduleValidationTest

20cognosce commented on src/test/java/school/faang/user_service/service/work_schedule/ServiceWorkScheduleTest.java:
```diff
@@ -0,0 +1,113 @@
+
+    @Test
+    public void testAddWorkSchedule() {
+        user.setId(userId);
+        user.setWorkSchedule(workSchedule);
```
сразу создавай юзера со всеми полями, не в тесте

20cognosce commented on src/test/java/school/faang/user_service/service/work_schedule/ServiceWorkScheduleTest.java:
```diff
@@ -0,0 +1,113 @@
+
+    @Test
+    public void testUpdateWorkScheduleDto() {
+        user.setId(userId);
+        user.setWorkSchedule(workSchedule);
```
почему одно и то же в каждом тесте?
